### ISSUE TITLE
Remove dependency on Gulp

### ISSUE THREAD

**Issue by felixfbecker** [sequelize/cli#344]
There is no reason why this depends on Gulp. Gulp brings in a lot of dependencies like vinyl-fs and in turn graceful-fs, which we dont need at all. All Gulp is used for is task registration, which orchestrator (undertaker for gulp 4) are used for. 

Using yargs would do the job _much_ better.

**Comment by davidgoli** [comment]
👍

**Comment by sdepold** [comment]
I agree. Back then I thought it would be a funny an easy going idea but it turned out to be a nightmare instead. I'm all for it!

**Comment by davidgoli** [comment]
At the very least a short-term fix would involve upgrading gulp to silence the `graceful-fs` deprecation messages.

**Comment by tdreyno** [comment]
Not to mention this! https://snyk.io/test/npm/sequelize-cli

**Comment by felixfbecker** [comment]
@tdreyno This is a CLI tool, not something you expose via HTTP or something, and sequelize-cli doesn't use the minimatch globbing at all. There is no vulnerability here.

**Comment by tdreyno** [comment]
Fair. I'm using automated security checking tools for node and it's complaining because of that, regardless of if the vector is real or not.

**Comment by andrew-cunliffe** [comment]
This is a pain, we have been asked to run `nsp check` as part of our CI build, but this dependency is the only one flagging a problem and its a false positive on the check :(

:+1 to remove gulp as a required dependency

**Comment by felixfbecker** [comment]
nsp allows to exclude certain packages iirc. Especially since sequelize-cli is a devdep...

**Comment by eddiejaoude** [comment]
`sequelize-cli` is a **Dev Dep**, doesn't state that in their docs. 

<img width="994" alt="screenshot 2016-11-28 15 08 56" src="https://cloud.githubusercontent.com/assets/624760/20673430/07977fbc-b57d-11e6-96ad-30d6050799fe.png">

They should add `--save-dev` 😞

**Comment by andrew-cunliffe** [comment]
So, I just tried our code without this dependency and migrations don't work now....

```
Running "shell:migrateUp" (shell) task
Migrating blog service up in test mode
{ [String: '']
  stdout: '',
  stderr: '/bin/sh: ./node_modules/.bin/sequelize: No such file or directory\n',
  code: 127,
  cat: [Function: bound ],
  head: [Function: bound ],
  tail: [Function: bound ],
  to: [Function: bound ],
  toEnd: [Function: bound ],
  sed: [Function: bound ],
  sort: [Function: bound ],
  uniq: [Function: bound ],
  grep: [Function: bound ],
  exec: [Function: bound ] }
Warning: Done, with errors: command "./scripts/migrations.js migrate" (target "migrateUp") exited with code 1. Use --force to continue.
```

What is the recommended way to run migrations on deployment if this is a dev dependency?

While I understand it might not be part of the running production code we need to execute the dependency on the servers to run the migrations, this means we need to be happy it has no vulnerabilities?

Open to your thoughts / ideas on this

**Comment by felixfbecker** [comment]
My understanding that `devDependencies` are everything needed to build and test your code. For example also Babel or TypeScript. In your production env you of course also need to install those to build your project, so I would never use `--production` flag.

**Comment by blacksonic** [comment]
Created pull request that fixes it https://github.com/sequelize/cli/pull/430

**Comment by blacksonic** [comment]
@felixfbecker Isn't it used only for testing? I don't see any sign of it in the bin file.

**Comment by idris** [comment]
Would an upgrade to Gulp 4 fix this issue?  Especially in the 2.x.x version of `sequelize-cli`

/cc @sushantdhiman

**Comment by webmobiles** [comment]
why use one thing old like Gulp, is not better webpack ? why sequelize tool don't generate ES6 js ? i am newbie with sequelize, but i don't understand why to write code like 5 years ago

**Comment by sushantdhiman** [comment]
Because this project is around 7 years old, be patient we are working on fixing things. If you want things to accelerate send me a PR :)

**Comment by webmobiles** [comment]
oh my God,; 7 years old :-) , must too look another ES6/ES7 Orm

**Comment by sushantdhiman** [comment]
@webmobiles I suggest https://github.com/Vincit/objection.js/ , you are welcome to try Sequelize any time :). Its open source after all

**Comment by webmobiles** [comment]
thanks susan , it was a joke, i am a dinosaur that comes from old php 4, in the new node world, i feel like i'm not working with modern tools so that gets me crazy, thanks , hope  to learn to contribute to modernize all these old packages
